### DISCUSSION TITLE
How to learn how to develop new project type?

### DISCUSSION THREAD

**Discussion by gherlein** [projen/projen#2215] [General]
I want to be able to define an additional project type that builds out the folder contents with even more files.  Instead of just a bare typescript app, I want to create a "telephony app" that is pre-set for all the things to build an application using the Amazon Chime SDK PSTN audio to build a basic telephony application.  I want to drop in all the code for this to reduce the undifferentiated heavy lifting of building the plumbing that is mostly stock code.  How do I learn how the "project types" work without having to grok ALL the internals of projen?

**Comment by Chriscbr** [comment]
Right now I don't think we have any docs about setting up new project types, this is a definitely a missing area of support. In the meantime I think there are some blog posts our community has written about this ([1](https://kennethwinner.com/2021/03/07/projen-external-module-github/), [2](https://itnext.io/projen-nodejs-project-boilerplating-15c20665c6b5)) - can you try taking a look and see if these help?

From your use case it sounds like you might want to create a project type that extends `TypeScriptProject` and which bundles some code using the `SampleFile` or `SampleDir` components. The API reference at https://projen.io lists some of the available options for configuring these.

**Comment by pgollucci** [comment]
I can help you with this.  I'll create a md doc file for projen.io. Ping me on cdk-dev slack in #projen

**Comment by gherlein** [comment]
Thanks for the links, it gives me a start.  I can help do stuff, document, test, review.  My amazon username is also gherlein so I'm easily findable.

**Comment by thoroc** [comment]
Did you ever got around doing that?
